bundles/framework/publisher2/handler/MapLayersHandler.js

```diff
@@ -0,0 +1,198 @@
+            layers: layers,
+            baseLayers: baseLayers
+        });
+    }
+
+    openLayerList () {
+        this.sandbox.postRequestByName(
+            'ShowFilteredLayerListRequest',
+            ['publishable', true]
+        );
+    }
+
+    addBaseLayer (layer) {
+        this.plugin.addBaseLayer(layer);
+        this.getLayers();
+    }
+
+    removeBaseLayer (layer) {
+        this.plugin.removeBaseLayer(layer);
+        this.getLayers();
+    }
+
+    /**
+     * @private @method _getToolPluginMapfullConf
+     * Get map view cofiguration (from mapfull) for this tool
+     * @return {Object / null} config or null if not found
+     */
+    getToolPluginMapfullConf () {
+        const { configuration } = this.data || {};
+        if (!configuration) {
+            return null;
+        }
+        const { mapfull = {} } = configuration;
+        const { conf = {} } = mapfull;
+        const { plugins = [] } = conf;
+        // data.configuration.mapfull.conf.plugins
+        return plugins.find(plug => plug.id === TOOL_ID);
+    }
+
+    startPlugin () {
+        const config = {
+            baseLayers: this.pluginConf?.config?.baseLayers || [],
+            defaultBaseLayer: this.pluginConf?.config?.defaultBaseLayer || this.pluginConf?.config?.baseLayers[0] || null,
+            showMetadata: this.pluginConf?.config?.showMetadata || false
+        };
+
+        const plugin = Oskari.clazz.create(TOOL_ID, config);
+        this.mapModule.registerPlugin(plugin);
+
+        plugin.startPlugin(this.sandbox);
+        return plugin;
+    }
+
+    stopPlugin () {
+        if (this.plugin) {
+            this.setShowMetadata(false);
+            if (this.sandbox) {
+                this.plugin.stopPlugin(this.sandbox);
+            }
+            this.mapModule.unregisterPlugin(this.plugin);
+        }
+    }
+
+    createEventHandlers () {
+        const handlers = {
+            /**
+             * @method AfterMapLayerAddEvent
+             * @param {Oskari.mapframework.event.common.AfterMapLayerAddEvent} event
+             *
+             * Updates the layerlist
+             */
+            AfterMapLayerAddEvent: function (event) {
+                this.getLayers();
+            },
+
+            /**
+             * @method AfterMapLayerRemoveEvent
+             * @param {Oskari.mapframework.event.common.AfterMapLayerRemoveEvent} event
+             *
+             * Updates the layerlist
+             */
+            AfterMapLayerRemoveEvent: function (event) {
+                this.getLayers();
+            },
+            /**
+             * @method AfterRearrangeSelectedMapLayerEvent
+             * @param {Oskari.mapframework.event.common.AfterRearrangeSelectedMapLayerEvent} event
+             *
+             * Updates the layerlist
+             */
+            AfterRearrangeSelectedMapLayerEvent: function (event) {
+                if (event._creator !== this.getName() && event._fromPosition !== event._toPosition) {
```

Comment:
Layers can no longer be arranged by this tool so event._creator should never be this.getName() -> we should be able to remove the event._creator check.